custom_components/ical/calendar.py

```diff
@@ -73,11 +75,12 @@ async def async_update(self):
         if event is None:
             self._event = event
             return
-        event = calculate_offset(event, OFFSET)
+        (summary, offset) = extract_offset(event["summary"], OFFSET)
+        event["summary"] = summary
+        self._offset_reached = is_offset_reached(get_date(event["start"]), offset)
```

Comment:
get_date() expects a dictionary, while we event|"start"] is a datetime.datetime. Did you mean
`is_offset_reached(event["start"], offset)`
?

Reply:
Thanks, I'll update the PR